tests/test.py

```diff
@@ -0,0 +1,33 @@
+import os
+import tempfile
+from tutor.types import Config
+from tutor.jobs import BaseJobRunner
+from tutor import config as tutor_config
+from tutor.commands.context import Context
+from tutor.commands.context import BaseJobContext
+
+
+CONTEXT = Context(os.path.join(tempfile.gettempdir(), "tutor"))
```

Comment:
This is causing a single context to be used to all unit tests. On my machine this is `/tmp/tutor`. Instead, a different tutor root should be created for all tests, with a randomly generated name so as not to overwrite existing tutor roots. So this global variable should be converted to a `def make_context()` function. Also, this test context should really be temporary and be deleted after each test run.

Reply:
The main reason I made single static context is to prevent the work needed to scaffold each unique context root and cleanup (i.e. remove) it after the single test is done to keep the test performance time to as minimum as possible. But if that is not a concern for you then I'm find with your idea to make unique context for each test.

Reply:
The changes has been made into [`helpers.py`](https://github.com/overhangio/tutor/blob/f4f5d6001167341913688f43215b89ac013f84e6/tests/helpers.py) to ensure all TestContext are created with new unique root directory for each test case.

Usage is as follows:
``` python
    def test_load_yml_config(self) -> None:
        with TestContext() as context:
            config = context.load_config()
            current = tutor_config.load(context.root)
            self.assertEqual(config, current)
```

Due to the scaffolding and cleanup for the TestContext() for each test case, the overall unit test time has increased until around 35seconds:
![image](https://user-images.githubusercontent.com/10910968/144938241-c2d07705-3427-4c23-a436-5e9457ed50f4.png)